naga/src/back/spv/block.rs

```diff
@@ -2693,6 +2758,6 @@ impl<'w> BlockContext<'w> {
         };
 
         self.function.consume(block, termination);
-        Ok(())
+        Ok(true)
```

Comment:
nitpick(non-blocking): I am concerned that while it was easiest to author this PR with `Ok(true)`, the easiest thing to read and verify later is likely to be a form where we are forced to decide per variant what the answer to the question of "do we expect control flow to connect to the end of the block".

suggestion: Let's change this code to use a forced assignment of the "end of block" control flow query in each variant of `Statement` matching.

I'm happy to make the change myself, if initial effort is the only objection.

Reply:
I don't think that would be an improvement. For readers, that's effectively adding a note to most arms of this match saying, "This concern, which you might not have been aware of, isn't relevant here."

More generally: Being aware of all concerns all the time would certainly eliminate bugs, but it's not a feasible way for humans to operate. Helping people ignore what's not relevant is the name of the game.

Your earlier suggestion actually shows a much better way to address this concern. If we introduce a new enum to use in the function's return type, then a contributor cannot add an early `return` (thus skipping the `exit` processing) without having to consider which enum variant to return. This forces them to consider the question at exactly and only the points where it becomes relevant.

Reply:
I think my other commit history makes it clear that, if I'm trying to optimize for ease of authoring, I'm pretty bad at that. 

But what I hear you saying is, this PR was a pain to make sense of.